### PR TITLE
tree-wise: several cleanups for logging

### PR DESCRIPTION
**Pull request by yuwata**
- drop unnecessary SYNTHETIC_ERRNO() when the logger does not propagate error code,
- drop unnecessary '%m' in error message when the error code is specified with SYNTHETIC_ERRNO(),
- add missing full stop at the end of log message,
- use RET_GATHER().

### CODE REVIEW

File: src/core/unit.c [hunk 1]
```diff
@@ -243,7 +243,7 @@ int unit_add_name(Unit *u, const char *text) {
         if (unit_name_is_valid(text, UNIT_NAME_TEMPLATE)) {
                 if (!u->instance)
                         return log_unit_debug_errno(u, SYNTHETIC_ERRNO(EINVAL),
-                                                    "instance is not set when adding name '%s': %m", text);
+                                                    "instance is not set when adding name '%s'.", text);
```
Comment (YHNdnzj) [inline src/core/unit.c]: If you touch this, also uppercase the first letter? (and below)

File: src/cryptsetup/cryptsetup.c [hunk 2]
```diff
@@ -523,7 +523,7 @@ static int parse_one_option(const char *option) {
                  * where %<key_type> is optional and defaults to 'user'.
                  */
                 if (!(sep = strstr(val, "::")))
-                        return log_error_errno(SYNTHETIC_ERRNO(EINVAL), "Failed to parse link-volume-key= option value: %m");
+                        return log_error_errno(SYNTHETIC_ERRNO(EINVAL), "Failed to parse link-volume-key= option value");
```
Comment (YHNdnzj) [inline src/cryptsetup/cryptsetup.c]: Maybe log about the value here?

File: src/cryptsetup/cryptsetup.c [hunk 3]
```diff
@@ -544,7 +544,7 @@ static int parse_one_option(const char *option) {
                 if (*sep == '%') {
                         /* must be separated by colon */
                         if (!(c = strchr(sep, ':')))
-                                return log_error_errno(SYNTHETIC_ERRNO(EINVAL), "Failed to parse link-volume-key= option value: %m");
+                                return log_error_errno(SYNTHETIC_ERRNO(EINVAL), "Failed to parse link-volume-key= option value");
```
Comment (YHNdnzj) [inline src/cryptsetup/cryptsetup.c]: ditto

File: src/pcrlock/pcrlock.c [hunk 4]
```diff
@@ -594,7 +594,7 @@ static int event_log_record_extract_firmware_description(EventLogRecord *rec) {
                         break;
 
                 default:
-                        log_warning_errno(SYNTHETIC_ERRNO(EBADMSG), "Unexpected separator payload %" PRIu32 ".", val);
+                        log_warning("Unexpected separator payload %" PRIu32 ".", val);
```
Comment (YHNdnzj) [inline src/pcrlock/pcrlock.c]: , ignoring?

File: src/shared/condition.c [hunk 5]
```diff
@@ -1101,15 +1101,15 @@ static int condition_test_psi(Condition *c, char **env) {
          * smaller or equal to the current pressure average over 5 minutes. */
         r = extract_many_words(&value, "/", 0, &third, &fourth);
         if (r <= 0)
-                return log_debug_errno(r < 0 ? r : SYNTHETIC_ERRNO(EINVAL), "Failed to parse condition parameter %s: %m", c->parameter);
+                return log_debug_errno(r ?: SYNTHETIC_ERRNO(EINVAL), "Failed to parse condition parameter %s: %m", c->parameter);
```
Comment (YHNdnzj) [inline src/shared/condition.c]: I'd like to keep the explicit comparison here

**Comment by yuwata** [comment]
@YHNdnzj Thank you for the review. Addressed all suggestions. Upgrading the green label.

**Pull request merged**